crates/ere-hosts/src/main.rs

```diff
@@ -165,39 +165,39 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     {
         run_cargo_patch_command("sp1")?;
         let sp1_zkvm = new_sp1_zkvm(resource.clone())?;
-        run_benchmark_ere("sp1", sp1_zkvm, action, &corpuses);
+        run_benchmark_ere("sp1", sp1_zkvm, action, &corpuses)?;
```

Comment:
The reason for why `run_benchmark_ere` can fail:
- Writing the `hardware.json` failed.
- Writing benchmark json outputs failed.

Both cases shouldn't fall into the "panic handling" logic. (i.e., in the previous way we did that, if writing the json outputs failed that [was considered a crash](https://github.com/eth-act/zkevm-benchmark-workload/blob/master/crates/benchmark-runner/src/lib.rs#L106-L107) which isn't strictly correct, so improving this a bit more now)